Program.cs

```diff
@@ -52,33 +52,31 @@ static void Main(string[] args)
                     cla.LocalPort = 0;
                 }
 
-                socket.Bind(new IPEndPoint(IPAddress.Any, cla.LocalPort));
-
-                P2pEndPoint p2pEndPoint = GetExternalEndPoint(socket);
-
-                if (p2pEndPoint == null)
-                    return;
-
-                Console.WriteLine("Tell this to your peer: {0}", p2pEndPoint.External.ToString());
+                InitializeSockets();
 
-                Console.WriteLine();
                 Console.WriteLine();
 
-                Console.Write("Enter the ip:port of your peer: ");
-                string peer = Console.ReadLine();
+                Console.Write("Enter the ip of your peer: ");
+                var remoteIp = Console.ReadLine();
 
-                if (string.IsNullOrEmpty(peer))
+                if (string.IsNullOrEmpty(remoteIp))
                 {
-                    Console.WriteLine("Invalid ip:port entered");
+                    Console.WriteLine("Invalid ip entered");
                     return;
                 }
 
-                // try again to connect to external to "reopen" port
-                GetExternalEndPoint(socket);
+                Console.Write("Enter the ports of your peer (comma separate): ");

```

Comment:
why change the UX of the program? you made the user have to do an extra copy+paste operation; but I think it's not needed: you should only ask about the IP and ports once, but instead of the format being ip:port, the format will be ip:port,port,port,..,port

Reply:
OK